src/main/java/juja/sqlcmd/InMemoryDatabaseManager.java

```diff
@@ -0,0 +1,127 @@
+package juja.sqlcmd;
+
+import java.sql.SQLException;
+
+public class InMemoryDatabaseManager implements DatabaseManager {
+    private Table[] tables;
+
+    public InMemoryDatabaseManager() {
+        tables = new Table[0];
+    }
+
+    public void createTable(String tableName, int columns_number) {
+        if (tableName == null) throw new IllegalArgumentException("Table name must be not null!");
+
+        if (columns_number < 0) throw new IllegalArgumentException("Number of columns must be positive!");
+
+        Table[] tmp = new Table[tables.length + 1];
+        System.arraycopy(tables, 0, tmp, 0, tables.length);
+        tables = tmp;
+        int lastElementIndex = tables.length - 1;
+        tables[lastElementIndex] = new Table(tableName, columns_number);
+    }
+
+    @Override
+    public boolean connect(String database, String user, String password) {
+        return true;
+    }
+
+    @Override
+    public String[] getTableNames() throws SQLException {
+        int numberOfTables = tables.length;
+        String[] tableNames = new String[numberOfTables];
+        for (int i = 0; i < numberOfTables; i++) {
+            Table currentTable = tables[i];
+            tableNames[i] = currentTable.getTableName();
+        }
+        return tableNames;
+    }
+
+    @Override
+    public DataSet[] getTableData(String tableName) throws SQLException {
+        Table table = tableByName(tableName);
+        return table != null ? table.tableData() : new DataSet[0];
+    }
+
+    @Override
+    public boolean insert(String tableName, DataSet dataset) {
+        Table table = tableByName(tableName);
+        return table != null && table.insert(dataset);
+    }
+
+    @Override
+    public boolean delete(String tableName, int id) {
+        Table table = tableByName(tableName);
+        return table != null && table.delete(id);
+    }
+
+    @Override
+    public void close() throws SQLException {
+        //NOP
+    }
+
+    private Table tableByName(String tableName) {
+        for (Table currentTable : tables) {
+            if (currentTable.getTableName().equals(tableName)) {
+                return currentTable;
+            }
+        }
+        return null;
+    }
+
+    private static class Table {
```

Comment:
This implementation has many operation arraycopy
I think, we could repeat idea from ArrayList implementation with initial capacity.

You can change it if you want but  for fun only :)

Reply:
Yup! You're right, I was thinking about that, but decided to make this implementation easier... I even was thinking to provide user identification))) Maybe in future. That's interesting. Thanks for your comments!